src/validation.cpp

```diff
@@ -2880,7 +2880,7 @@ std::vector<unsigned char> GenerateCoinbaseCommitment(CBlock& block, const CBloc
     std::vector<unsigned char> commitment;
     int commitpos = GetWitnessCommitmentIndex(block);
     std::vector<unsigned char> ret(32, 0x00);
-    if (consensusParams.vDeployments[Consensus::DEPLOYMENT_SEGWIT].nTimeout != 0) {
+    if (gArgs.GetArg("-segwitheight", consensusParams.SEGWITHeight) >= 0) {
```

Comment:
why do you need to check the `-segwitheight` argument here? Hasn't `consensusParams.SEGWITHeight` been updated by the time this function is called?

Reply:
I try to keep the current behavior that the witness commitment is not made if segwit is disabled (before: nTimeout == 0 / after: SEGWITHeight < 0). If you use `-segwitheight=-1`, it should work identical to a non-segwit node. This is needed for some tests in p2p-segwit.py related to the witness commitment. (we could remove those tests, of course)

Reply:
ok. May I should use `consensusParams.SEGWITHeight != std::numeric_limits<int>::max()` instead

Reply:
sorry, I misunderstood. I think your existing change is fine (and I agree that we should remove this at a later date once the tests are updated)